File: ce/customer-engagement/on-premises/snippets/csharp/CRMV8/businessmanagement/cs/cancelsalesorder.cs
// <snippetcancelsalesorder>



 // Define columns to be retrieved after creating the order
 salesOrderColumns = new ColumnSet("salesorderid", "totalamount");

 // Convert the quote to a sales order
 convertQuoteRequest =
     new ConvertQuoteToSalesOrderRequest()
     {
         QuoteId = _quoteId,
         ColumnSet = salesOrderColumns
     };
 convertQuoteResponse =
     (ConvertQuoteToSalesOrderResponse)_serviceProxy.Execute(convertQuoteRequest);
 SalesOrder closeSalesOrder = (SalesOrder)convertQuoteResponse.Entity;
 _closeSalesOrderId = closeSalesOrder.Id;

 CancelSalesOrderRequest cancelRequest = new CancelSalesOrderRequest()
 {
     OrderClose = new OrderClose()
     {
         SalesOrderId = closeSalesOrder.ToEntityReference(),
         Subject = "Close Sales Order " + DateTime.Now
     },
     Status = new OptionSetValue(-1)
 };
 _serviceProxy.Execute(cancelRequest);

 Console.WriteLine("Canceled sales order");

// </snippetcancelsalesorder>

File: ce/customerengagement/on-premises/snippets/visualbasic/CRMV8/businessmanagementvb/vb/cancelsalesorder.vb
' <snippetcancelsalesorder>


 ' Define columns to be retrieved after creating the order
 salesOrderColumns = New ColumnSet("salesorderid", "totalamount")

 ' Convert the quote to a sales order
 convertQuoteRequest = New ConvertQuoteToSalesOrderRequest() With
                       {
                           .QuoteId = _quoteId,
                           .ColumnSet = salesOrderColumns
                       }
 convertQuoteResponse = CType(_serviceProxy.Execute(convertQuoteRequest), 
     ConvertQuoteToSalesOrderResponse)
 Dim closeSalesOrder As SalesOrder = CType(convertQuoteResponse.Entity, 
     SalesOrder)
 _closeSalesOrderId = closeSalesOrder.Id

 Dim cancelRequest As New CancelSalesOrderRequest() With
     {
         .OrderClose = New OrderClose() With
                       {
                           .SalesOrderId = closeSalesOrder.ToEntityReference(),
                           .Subject = "Close Sales Order " &amp; Date.Now
                       },
         .Status = New OptionSetValue(-1)
     }
 _serviceProxy.Execute(cancelRequest)

 Console.WriteLine("Canceled sales order")

' </snippetcancelsalesorder>